### PR TITLE
fix(link): added feature to stop click propagation for disabled link

### PR DESCRIPTION
**Pull request by Rajdeepc**
## Description
`disabled` attribute should not allow click propagation in `sp-link`
Added a logic to stopPropagation and prevent any clicks to bubble up from the element.

## Related issue(s)

- #4252 

## How has this been tested?

-   [x] _Test case 1_
    1. Go [here](https://link-disabled-click--spectrum-web-components.netlify.app/storybook/?path=/story/link--disabled)
    2. Try to TAB or click on the link.

## Types of changes

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [x] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

File: packages/link/src/Link.ts [hunk 1]
```diff
@@ -47,6 +47,22 @@ export class Link extends LikeAnchor(Focusable) {
...
+        this.addEventListener('click', this.handleClickCapture, {
+            capture: true,
+        });
+    }
+
+    private handleClickCapture(event: Event): void | boolean {
+        if (this.disabled) {
+            event.preventDefault();
+            event.stopImmediatePropagation();
+            event.stopPropagation();
```
Comment (Westbrook) [inline packages/link/src/Link.ts:61]: https://developer.mozilla.org/en-US/docs/Web/CSS/pointer-events#none

Comment (Rajdeepc) [reply packages/link/src/Link.ts]: Yes this was one of the simplest ways. I feel there are many places in the codebase where we are adding the above code to handle event propagation.

**Comment by Rajdeepc** [comment]
> I _think_ this "corrects" this functionally but I'm worried about it actually working while looking the same as a NOT disabled link. Can we get some design direction?

There is a comment on this by @KayiuCarlson on [this](https://adobedesign.slack.com/archives/CESK60MQD/p1713364564641149?thread_ts=1712614724.465019&cid=CESK60MQD) discussion thread as:

> Disabled state for [Link](https://spectrum.corp.adobe.com/page/link/) is not currently available and being considered in Spectrum. In order for Spectrum design to consider this request (and have it be properly captured to the design tokens), we'd need to understand the use case as well as making sure that there's accessibility consideration. Therefore, it's not just a simple color change. If you're working with a designer, please have them reach out to us at Spectrum design. If you're not working with a designer, we'd still want to see the design of which a disabled Link would like to be used and its context. Without that, this would be a custom change on the product side only.

**Review by Westbrook — APPROVED** [review]
Lets take this in. Rebasing to `main` and will merge on green.

**Pull request merged**